leakcanary-deobfuscation-gradle-plugin/src/main/java/com/squareup/leakcanary/deobfuscation/CopyObfuscationMappingFileTask.kt

```diff
@@ -0,0 +1,54 @@
+package com.squareup.leakcanary.deobfuscation
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.CacheableTask
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.PathSensitivity
+import org.gradle.api.tasks.SkipWhenEmpty
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+
+@CacheableTask
+open class CopyObfuscationMappingFileTask : DefaultTask() {
+
+  @Input
+  var variantDirName: String? = null
+
+  @Input
+  @SkipWhenEmpty
+  @PathSensitive(PathSensitivity.RELATIVE)
+  var mappingFile: File? = null
+
+  @Input
+  @SkipWhenEmpty
+  @PathSensitive(PathSensitivity.RELATIVE)
+  var mergeAssetsDirectory: File? = null
+
+  @get:OutputFile
+  @get:PathSensitive(PathSensitivity.RELATIVE)
+  val leakCanaryAssetsOutputFile: File
+    get() = File(mergeAssetsDirectory, "leakCanaryObfuscationMapping.txt")
+
+  init {
+    description = "Puts obfuscation mapping file in assets directory."
+  }
+
+  @TaskAction
+  fun copyObfuscationMappingFile() {
+    mappingFile?.let { mappingFile ->
```

Comment:
be more assertive here. And generally speaking we don't want a task in gradle that runs and silently does nothing.
Here:
```kotlin
 if mappingFile does not exist --> crash
 if mergeAssetsDirectory does not exists --> create
 if leakCanaryAssetsOutputFile exists --> delete 
copy.
 ```